Justfile for CosmWasm contracts

pitasi opened the pull request:
Adds a justfile for building our contracts, it also automatically deploys "ethereum-analyzer" as part of our localnet setup (requires Docker and `jq`)

closes #376 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added commands for managing and compiling CosmWasm contracts using Docker.
  - Introduced automatic deployment of contracts in the local development environment.
  
- **Improvements**
  - Enhanced default settings to ensure uninterrupted Ethereum transactions.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pitasi committed:
chore(contracts): add new contracts to Cargo.toml

pitasi committed:
chore: add justfile for compiling and deploying smart contracts

pitasi committed:
chore(localnet): deploy ethereum-analyzer contract

pitasi committed:
chore(spaceward): set default ethereum-analyzer contract for localnet

mn13 commented:
Tried locally just localnet
1. My version of sed is not working with `sed -i 's/cors_allowed_origins = \[\]/cors_allowed_origins = ["*"]/' ~/.warden/config/config.toml`, but working with
`sed -i '' -e 's/cors_allowed_origins = \[\]/cors_allowed_origins = ["*"]/' ~/.warden/config/config.toml`
I have Mac Sonoma 14.4.1, text_cmds-165.100.8 (it's sed version in MacOs, from here https://opensource.apple.com/releases/)

2. After fixing have error `unknown command "add-genesis-space" for "genesis"`

pitasi commented:
> Tried locally just localnet
> 
> 1. My version of sed is not working with `sed -i 's/cors_allowed_origins = \[\]/cors_allowed_origins = ["*"]/' ~/.warden/config/config.toml`, but working with
>    `sed -i '' -e 's/cors_allowed_origins = \[\]/cors_allowed_origins = ["*"]/' ~/.warden/config/config.toml`
>    I have Mac Sonoma 14.4.1, text_cmds-165.100.8 (it's sed version in MacOs, from here https://opensource.apple.com/releases/)

I fixed that issue on main, I now rebased this PR so it contains the fix :)

> 2. After fixing have error `unknown command "add-genesis-space" for "genesis"`

I think you have a really old version of wardend. You can update it with `just install`.
